distributed/tests/test_worker.py

```diff
@@ -2744,36 +2748,32 @@ async def test_acquire_replicas_many(c, s, *workers):
         await asyncio.sleep(0.001)
 
 
-@pytest.mark.slow
-@gen_cluster(client=True, Worker=Nanny)
-async def test_acquire_replicas_already_in_flight(c, s, *nannies):
+@gen_cluster(client=True, nthreads=[("", 1)])
+async def test_acquire_replicas_already_in_flight(c, s, a):
     """Trying to acquire a replica that is already in flight is a no-op"""
+    async with BlockedGetData(s.address) as b:
+        x = c.submit(inc, 1, key="x", workers=[b.address])
+        y = c.submit(inc, x, key="y", workers=[a.address])
+        await _wait_for_state("x", a, "flight")
 
-    class SlowToFly:
-        def __getstate__(self):
-            sleep(0.9)
-            return {}
+        s.request_acquire_replicas(a.address, ["x"], stimulus_id="test")
+        while a.log[-1][:5] != ("x", "flight", "fetch", "flight", {}):
+            await asyncio.sleep(0.01)
```

Comment:
Isn't there an event we can listen to? I consider events much more robust than the transition logs and much easier to read.

I would even prefer having a plain sleep in here than listening to the transition log

Reply:
Changed to trigger the event directly on the worker so that no wait is involved